optuna/terminator/terminator.py

```diff
@@ -0,0 +1,45 @@
+import abc
+from typing import Optional
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.terminator.erroreval import BaseErrorEvaluator
+from optuna.terminator.erroreval import CrossValidationErrorEvaluator
+from optuna.terminator.improvement.evaluator import BaseImprovementEvaluator
+from optuna.terminator.improvement.evaluator import DEFAULT_MIN_N_TRIALS
+from optuna.terminator.improvement.evaluator import RegretBoundEvaluator
+from optuna.trial import TrialState
+
+
+class BaseTerminator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def should_terminate(self, study: Study) -> bool:
+        pass
+
+
+@experimental_class("3.2.0")
+class Terminator(BaseTerminator):
+    def __init__(
+        self,
+        improvement_evaluator: Optional[BaseImprovementEvaluator] = None,
+        error_evaluator: Optional[BaseErrorEvaluator] = None,
+        min_n_trials: int = DEFAULT_MIN_N_TRIALS,
+    ) -> None:
+        self._improvement_evaluator = improvement_evaluator or RegretBoundEvaluator()
+        self._error_evaluator = error_evaluator or CrossValidationErrorEvaluator()
+        self._min_n_trials = min_n_trials
```

Comment:
How about raising a ValueError exception when `min_n_trials` is less than or equal to 0? Otherwise, `CrossValidationErrorEvaluator` may raise an AssertionError exception I guess.

Reply:
(Noticed we actually get a `ValueError` from the `RegretBoundEvaluator` if `min_n_trials` is `0`.)

Reply:
Thanks for your comments. A non-positive `min_n_trials` now raises `ValueError`.
https://github.com/optuna/optuna/pull/4405/commits/ce89ef0f72bf1265b08ff864ef3cae8b35ec488f